Script error - Update Check Runs

Thanks for putting this together, it's really useful. I've been experiencing a couple of (what I imagine are) permission errors related to the GitHub App configuration. Are there some additional permissions that need to be granted beyond those stated below?

```
default_permissions:
  contents: :write
  metadata: :read
```

This script: https://github.com/imjohnbo/ok-to-test/blob/474c02cafe14327d1a60febb7eb375f5935a430e/.github/workflows/integration.yml#L52-L73

Fails with this error for me:
```
2023-03-05T18:53:26.8384989Z RequestError [HttpError]: Resource not accessible by integration
2023-03-05T18:53:26.8387986Z     at /home/runner/work/_actions/actions/github-script/v6/dist/index.js:6803:21
2023-03-05T18:53:26.8388365Z     at processTicksAndRejections (node:internal/process/task_queues:96:5)
2023-03-05T18:53:26.8388932Z     at async eval (eval at callAsyncFunction (/home/runner/work/_actions/actions/github-script/v6/dist/index.js:15099:16), <anonymous>:13:26)
2023-03-05T18:53:26.8389463Z     at async main (/home/runner/work/_actions/actions/github-script/v6/dist/index.js:15198:20) {
2023-03-05T18:53:26.8389745Z   status: 403,
2023-03-05T18:53:26.8389922Z   response: {
2023-03-05T18:53:26.8390378Z     url: 'https://api.github.com/repos/GoodhandSolutions/terraform-snowflake-playground/check-runs/11775549491',
2023-03-05T18:53:26.8391081Z     status: 403,
2023-03-05T18:53:26.8391282Z     headers: {
2023-03-05T18:53:26.8391572Z       'access-control-allow-origin': '*',
2023-03-05T18:53:26.8392544Z       'access-control-expose-headers': 'ETag, Link, Location, Retry-After, X-GitHub-OTP, X-RateLimit-Limit, X-RateLimit-Remaining, X-RateLimit-Used, X-RateLimit-Resource, X-RateLimit-Reset, X-OAuth-Scopes, X-Accepted-OAuth-Scopes, X-Poll-Interval, X-GitHub-Media-Type, X-GitHub-SSO, X-GitHub-Request-Id, Deprecation, Sunset',
2023-03-05T18:53:26.8393279Z       connection: 'close',
2023-03-05T18:53:26.8393538Z       'content-encoding': 'gzip',
2023-03-05T18:53:26.8393865Z       'content-security-policy': "default-src 'none'",
2023-03-05T18:53:26.8394227Z       'content-type': 'application/json; charset=utf-8',
2023-03-05T18:53:26.8394548Z       date: 'Sun, 05 Mar 2023 18:53:26 GMT',
2023-03-05T18:53:26.8394962Z       'referrer-policy': 'origin-when-cross-origin, strict-origin-when-cross-origin',
2023-03-05T18:53:26.8395324Z       server: 'GitHub.com',
2023-03-05T18:53:26.8395714Z       'strict-transport-security': 'max-age=31536000; includeSubdomains; preload',
2023-03-05T18:53:26.8396063Z       'transfer-encoding': 'chunked',
2023-03-05T18:53:26.8396408Z       vary: 'Accept-Encoding, Accept, X-Requested-With',
2023-03-05T18:53:26.8396746Z       'x-content-type-options': 'nosniff',
2023-03-05T18:53:26.8397024Z       'x-frame-options': 'deny',
2023-03-05T18:53:26.8397350Z       'x-github-api-version-selected': '2022-11-28',
2023-03-05T18:53:26.8397678Z       'x-github-media-type': 'github.v3',
2023-03-05T18:53:26.8398039Z       'x-github-request-id': '86C4:60B2:2DCE608:5F0BA85:6404E526',
2023-03-05T18:53:26.8398348Z       'x-ratelimit-limit': '1000',
2023-03-05T18:53:26.8398628Z       'x-ratelimit-remaining': '980',
2023-03-05T18:53:26.8398911Z       'x-ratelimit-reset': '1678044246',
2023-03-05T18:53:26.8399189Z       'x-ratelimit-resource': 'core',
2023-03-05T18:53:26.8399470Z       'x-ratelimit-used': '20',
2023-03-05T18:53:26.8400037Z       'x-xss-protection': '0'
2023-03-05T18:53:26.8400235Z     },
2023-03-05T18:53:26.8400407Z     data: {
2023-03-05T18:53:26.8400810Z       message: 'Resource not accessible by integration',
2023-03-05T18:53:26.8403050Z       documentation_url: 'https://docs.github.com/rest/reference/checks#update-a-check-run'
2023-03-05T18:53:26.8403371Z     }
2023-03-05T18:53:26.8403527Z   },
2023-03-05T18:53:26.8403684Z   request: {
2023-03-05T18:53:26.8403938Z     method: 'PATCH',
2023-03-05T18:53:26.8404396Z     url: 'https://api.github.com/repos/GoodhandSolutions/terraform-snowflake-playground/check-runs/11775549491',
2023-03-05T18:53:26.8404722Z     headers: {
2023-03-05T18:53:26.8405028Z       accept: 'application/vnd.github.-preview+json',
2023-03-05T18:53:26.8405475Z       'user-agent': 'actions/github-script octokit-core.js/3.6.0 Node.js/16.16.0 (linux; x64)',
2023-03-05T18:53:26.8405836Z       authorization: 'token [REDACTED]',
2023-03-05T18:53:26.8406169Z       'content-type': 'application/json; charset=utf-8'
2023-03-05T18:53:26.8406411Z     },
2023-03-05T18:53:26.8406688Z     body: '{"status":"completed","conclusion":"success"}',
2023-03-05T18:53:26.8407009Z     request: { agent: [Agent], hook: [Function: bound bound register] }
2023-03-05T18:53:26.8407248Z   }
2023-03-05T18:53:26.8407404Z }
2023-03-05T18:53:26.8437830Z ##[error]Unhandled error: HttpError: Resource not accessible by integration
```

Similarly the dispatching action, raises warnings for: `Failed to set reaction on comment ID 1455172514.`

Are these just user errors on my part or is it something else?

PS. I think the action versions need to be updated to prevent warnings being raised about the Node version. I can raise a PR for that.

--- a/.github/workflows/integration.yml
+++ b/.github/workflows/integration.yml
@@ -14,22 +14,28 @@ jobs:
     steps:
 
     - name: Branch based PR checkout
-      uses: actions/checkout@v2
+      uses: actions/checkout@v3
 
     # <insert integration tests needing secrets>
 
   # Repo owner has commented /ok-to-test on a (fork-based) pull request
   integration-fork:
     runs-on: ubuntu-latest
-    if: 
+    permissions:
+      pull-requests: write
+      checks: write
+    if: |
       github.event_name == 'repository_dispatch' &&
-      github.event.client_payload.slash_command.sha != '' &&
-      contains(github.event.client_payload.pull_request.head.sha, github.event.client_payload.slash_command.sha)
+      github.event.client_payload.slash_command.args.named.sha != '' &&
+      contains(
+        github.event.client_payload.pull_request.head.sha,
+        github.event.client_payload.slash_command.args.named.sha
+      )
     steps:
 
     # Check out merge commit
     - name: Fork based /ok-to-test checkout
-      uses: actions/checkout@v2
+      uses: actions/checkout@v3
       with:
         ref: 'refs/pull/${{ github.event.client_payload.pull_request.number }}/merge'
 
@@ -39,7 +45,7 @@ jobs:
         echo "Integration tests... success! ;-)"
 
     # Update check run called "integration-fork"
-    - uses: actions/github-script@v5
+    - uses: actions/github-script@v6
       id: update-check-run
       if: ${{ always() }}
       env:

--- a/.github/workflows/ok-to-test.yml
+++ b/.github/workflows/ok-to-test.yml
@@ -8,13 +8,17 @@ on:
 jobs:
   ok-to-test:
     runs-on: ubuntu-latest
+    permissions:
+      pull-requests: write
     # Only run for PRs, not issue comments
     if: ${{ github.event.issue.pull_request }}
     steps:
     # Generate a GitHub App installation access token from an App ID and private key
     # To create a new GitHub App:
     #   https://developer.github.com/apps/building-github-apps/creating-a-github-app/
-    # See app.yml for an example app manifest
+    # See app.yml for an example app manifest - you can use dummy URLs when generating
+    # the app such as example.com urls. They do not need to properly resolve for the
+    # purpose that they app is being used for.
     - name: Generate token
       id: generate_token
       uses: tibdex/github-app-token@v1
@@ -23,7 +27,7 @@ jobs:
         private_key: ${{ secrets.PRIVATE_KEY }}
 
     - name: Slash Command Dispatch
-      uses: peter-evans/slash-command-dispatch@v1
+      uses: peter-evans/slash-command-dispatch@v3
       env:
         TOKEN: ${{ steps.generate_token.outputs.token }}
       with:
@@ -32,5 +36,4 @@ jobs:
         reaction-token: ${{ secrets.GITHUB_TOKEN }}
         issue-type: pull-request
         commands: ok-to-test
-        named-args: true
         permission: write

--- a/.github/workflows/ok-to-test.yml
+++ b/.github/workflows/ok-to-test.yml
@@ -16,7 +16,7 @@ jobs:
     # Generate a GitHub App installation access token from an App ID and private key
     # To create a new GitHub App:
     #   https://developer.github.com/apps/building-github-apps/creating-a-github-app/
-    # See app.yml for an example app manifest - you can use dummy URLs when generating
+    # See app.yml for an example app manifest - you can use fake URLs when generating
     # the app such as example.com urls. They do not need to properly resolve for the
     # purpose that they app is being used for.
     - name: Generate token

